src/ocaml/preprocess/parser_raw.mly

```diff
@@ -651,8 +651,7 @@ let expr_of_lwt_bindings ~loc lbs body =
   let default_loc = ref Location.none
 
   let default_expr () =
-    let id = Location.mkloc "merlin.hole" !default_loc in
-    Exp.mk ~loc:!default_loc (Pexp_extension (id, PStr []))
+    Exp.mk ~loc:!default_loc (Pexp_hole)
```

Comment:
I don't think I like that diff, `default_expr` is inserted in place of expression nodes we couldn't parse. So if you insert `Pexp_hole`s the result of the `holes` command will contain garbage on files that don't parse.

I agree that these nodes and `(??)` inserted by the user were indistinguishable before, but now they probably should be?

Reply:
I pushed a commit (a955e71) reverting the changes to the recovered "holes".
They are represented by `merlin.hole` extension points as before.